assembl/static/js/app/views/messageColumnsPanel.js

```diff
@@ -0,0 +1,451 @@
+        return;
+      }
+      that.template = that.message_template;
+      that.render();
+    });
+  },
+
+  getGroupState: function() {
+    return this.basePanel.getGroupState();
+  },
+
+  getContainingGroup: function() {
+    return this.basePanel.getContainingGroup();
+  },
+
+  unblockPanel: function() {
+    this.basePanel.unblockPanel();
+  },
+
+  /**
+   * Synchronizes the panel with the currently selected idea (possibly none)
+   */
+  syncWithCurrentIdea: function() {
+    var currentIdea = this.getGroupState().get('currentIdea');
+    this.setCurrentIdea(currentIdea);
+  },
+
+  serializeData: function() {
+    var data = BaseMessageColumnView.prototype.serializeData.apply(this, arguments);
+    _.extend(data, {
+      column: this.model,
+      numColumns: this.model.collection.length,
+      canPost: Ctx.getCurrentUser().can(Permissions.ADD_POST),
+      color: this.model.get('color') || 'black',
+    });
+    return data;
+  },
+
+  processIsEnded: function() {
+    // heuristic: process is ended if header has content.
+    var header = this.model.get('header');
+    return header != undefined && header.length > 0;
+  },
+
+  onRender: function() {
+    if (this.isViewDestroyed()) {
+      return;
+    }
+    BaseMessageColumnView.prototype.onRender.apply(this, arguments);
+    var that = this,
+        canEdit = Ctx.getCurrentUser().can(Permissions.ADMIN_DISCUSSION),
+        renderId = _.clone(this._renderId);
+
+    if (this.processIsEnded() || canEdit) {
+      this.messageColumnDescription.show(new CKEditorField({
+        model: this.model,
+        modelProp: 'header',
+        canEdit: canEdit,
+      }));
+    }
+    this.renderMessageListViewStyleDropdown();
+    Promise.join(this.messagesIdsPromise, this.translationDataPromise, function(resultMessageIdCollection, translationData) {
+      if (that.isViewDestroyed()) {
+        return;
+      }
+
+      if (renderId != that._renderId) {
+        console.log("messageList:onRender() structure collection arrived too late, this is render %d, and render %d is already in progress.  Aborting.", renderId, that._renderId);
+        return;
+      }
+
+      that.destroyAnnotator();
+
+      that.ui.messageCount.html(i18n.sprintf(
+        i18n.gettext("%d “%s” messages"),
+        resultMessageIdCollection.length,
+        that.model.get('name').bestValue(translationData)));
+
+      //Some messages may be present from before
+      that.ui.messageFamilyList.empty();
+      that.clearRenderedMessages();
+
+      that.render_real();
+      that.ui.panelBody.scroll(function() {
+        var msgBox = that.$('.messagelist-replybox').height(),
+        scrollH = $(this)[0].scrollHeight - (msgBox + 25),
+        panelScrollTop = $(this).scrollTop() + $(this).innerHeight();
+
+        //This event cannot be bound in ui, because backbone binds to
+        //the top element and scroll does not propagate
+        that.$(".panel-body").scroll(that, that.scrollLogger);
```

Comment:
This style was copied from messageList, I did not question it. If you want to improve it, feel free but do it in both files.